File: Java/ToyLanguageGUI_exam_final/src/utils/ExecStack.java
package utils;
import java.util.Deque;
import java.util.ArrayDeque;

public class ExecStack<T> implements IExecStack<T> {
    private Deque<T> stack;
    public ExecStack(){
        stack = new ArrayDeque<>();
    }
    public void push(T e){
        stack.push(e);
    }

    public T pop(){
        return stack.pop();
    }

    public boolean isEmpty(){ return stack.size() == 0; }

    @Override
    public String toString(){
        StringBuffer str = new StringBuffer();
        for(T e : stack)
            str.append(e + " \n");
        return str.toString();
    }

    @Override
    public Iterable<T> getAll() {
        return stack;
    }
}

File: ToyInterpretor/ToyInterpretor/Utils/ExecStack.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ToyInterpretor.Utils
{
    class ExecStack<T> : IExecStack<T>
    {
        private Stack<T> stack;
        public ExecStack()
        {
            stack = new Stack<T>();
        }
        public void push(T e)
        {
            stack.Push(e);
        }

        public T pop()
        {
            return stack.Pop();
        }

        public Boolean isEmpty() { return stack.Count() == 0; }

        
        public override String ToString()
        {
            StringBuilder str = new StringBuilder();
            foreach (T e in stack)
                str.Append(e + " \n");
            return str.ToString();
        }

        
        public IEnumerable<T> getAll()
        {
            return stack;
        }
    }
}
